### ISSUE TITLE
Shiny reset everything to default when click on shinyFile's shinyDirButton button.

### ISSUE THREAD

**Issue by darkwish121** [thomasp85/shinyFiles#33]
While building a shiny app, there is a button to choose directory of files this button made by shinyFiles package. There are also some radio buttons on page or a button to show/hide **Hide data chunk** a div. Problem is after first time select the folder, then select any other radio button which was not default or click on show/hide button it becomes **Show data chunk**, then as soon as second time click on **choose directory** button it reset radio button to default selected and change the value of show/hide button to default which was at first time load page.

Why it reset these elements by click on second time to folder select?

Here is the code.

**ui.R**

```
library(shiny)
library(shinyFiles)
library(shinyjs)

shinyUI(fluidPage(
  titlePanel("Data Insights"),

  sidebarPanel(
    useShinyjs(),
    shinyDirButton('directory', 'Folder select', 'Please select a folder'),

   tags$div(id = "DrawChartFiltersWraper",
      uiOutput("chartsTypeControls")

    ),
  ),
  mainPanel(
      uiOutput('allStats'),
      tags$br(),
      uiOutput('dataChunkHeading')
  ),
))
```

**server.R**

```
shinyServer(function(input, output, session) {

# make data frame from CSV files.
dataFrame <- reactive({

  volumes <- getVolumes()
  shinyDirChoose(input,
               'directory',
               roots=volumes,
               session=session,
               restrictions=system.file(package='base'))

  validate (
    need(input$directory != "", "<< Please select a CSV folder"),
    errorClass = "missing-folder"
  )


  getDataFrame(parseDirPath(volumes, input$directory))

})


output$chartsTypeControls <- renderUI({

  chartTypes <- c("Bar Chart", "Pie Chart", "Scatterplot")
  radioButtons("chartTypes", "Choose Chart Type", chartTypes)
})


# data Chunk
output$dataChunkHeading <- renderUI({
  toggleDataChunk <- actionButton("toggleDataChunk", "Hide data chunk", class = "btn-primary")
  HTML(paste("<h3>Data chunk of provided data.</h3>", toggleDataChunk))
})

output$allStats <- renderUI({

    total_rows = nrow(dataFrame)
    total_columns = ncol(dataFrame)
    overAllTxt <- paste('<h3>Overall Insights</h3><br /><p>', "Total Rows:", total_rows, "Total Columns:", total_columns, "</p>")
    HTML(overAllTxt)
  })
})
```

**Comment by burgerga** [comment]
Minimal example:

```
library(shiny)
library(shinyFiles)

# Define UI for application that draws a histogram
ui <- shinyUI(fluidPage(

   # Application title
   titlePanel("shinyDirButton resets select"),

   # Sidebar with a slider input for number of bins
   sidebarLayout(
      sidebarPanel(
       shinyDirButton("dir", "Set directory...", "Select directory"),
       selectInput("select1", "Regular select", choices = c("Choose one" = "", 1, 2)),
       uiOutput("render")
      ),

      # Show a plot of the generated distribution
      mainPanel(
      )
   )
))

# Define server logic required to draw a histogram
server <- shinyServer(function(input, output, session) {
  roots <- getVolumes()
  shinyDirChoose(input, 'dir', session=session, roots = roots)

  output$render <- renderUI(
    selectInput("select2", "Select via renderUI", choices = c("Choose one" = "", 1, 2))
  )
})

# Run the application
shinyApp(ui = ui, server = server)
```
As you can see it happens only for the selectInput added with renderUI

**Comment by burgerga** [comment]
Duplicate of https://github.com/thomasp85/shinyFiles/issues/23

**Comment by thomasp85** [comment]
closed by 1c82b5e4c67565afc740cf925c9eb98962db7e6d
